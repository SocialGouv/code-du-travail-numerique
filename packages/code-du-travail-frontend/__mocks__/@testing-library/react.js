/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
