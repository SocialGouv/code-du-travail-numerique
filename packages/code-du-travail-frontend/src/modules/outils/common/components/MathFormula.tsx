import "katex/dist/katex.min.css";
import TeX from "@matejmazur/react-katex";
import React from "react";

const asciiMathToTex = (ascii) => {
  // multiplications
  let tex = ascii.replace(/\*/g, `\\times`);
  // divisions
  const parenthesisMatcher = `(\\(.+\\))`;
  const nonParenthesisMatcher = `([^()\\s]+)`;
  const fullMatcher = `(?:${parenthesisMatcher}|${nonParenthesisMatcher})`;
  while (tex.includes("/")) {
    tex = tex.replace(
      new RegExp(`${fullMatcher}\\s?\\/\\s?${fullMatcher}`, "gm"),
      `\\frac{$1$2}{$3$4}`
    );
  }
  tex = tex.replace(new RegExp(`\%`, "gm"), `\\%`);
  return tex;
};

type Props = {
  formula: string;
};

export const MathFormula = ({ formula }: Props) => {
  return (
    <TeX
      title={formula.replace(/\*/g, `x`)}
      block
      style={{ width: "fit-content", overflow: "auto" }}
    >
      {asciiMathToTex(formula)}
    </TeX>
  );
};
