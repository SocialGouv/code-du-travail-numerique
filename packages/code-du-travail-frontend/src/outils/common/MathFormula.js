import TeX from "@matejmazur/react-katex";
import PropTypes from "prop-types";
import React from "react";

export const asciiMathToTex = (ascii) => {
  // multiplications
  let tex = ascii.replace(/\*/g, `\\times`);
  // divisions
  /* eslint-disable no-useless-escape */
  const parenthesisMatcher = `(\\(.+\\))`;
  const nonParenthesisMatcher = `([^()\\s]+)`;
  const fullMatcher = `(?:${parenthesisMatcher}|${nonParenthesisMatcher})`;
  while (tex.includes("/")) {
    tex = tex.replace(
      new RegExp(`${fullMatcher}\\s?\\/\\s?${fullMatcher}`, "gm"),
      `\\frac{$1$2}{$3$4}`,
    );
  }
  tex = tex.replace(new RegExp(`\%`, "gm"), `\\%`);
  return tex;
};

export const MathFormula = ({ formula }) => {
  return (
    <TeX
      title={formula.replace(/\*/g, `x`)}
      block
      style={{ width: "fit-content", overflow: "auto" }}
    >
      {asciiMathToTex(formula)}
    </TeX>
  );
};

MathFormula.propTypes = {
  formula: PropTypes.string.isRequired,
};
