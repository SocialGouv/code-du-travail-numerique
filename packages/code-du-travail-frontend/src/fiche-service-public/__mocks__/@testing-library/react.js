import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
