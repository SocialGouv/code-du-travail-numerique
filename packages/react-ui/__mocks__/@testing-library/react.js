import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
