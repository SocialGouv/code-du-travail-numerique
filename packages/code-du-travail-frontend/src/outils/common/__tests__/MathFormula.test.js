import { render } from "@testing-library/react";
import React from "react";

import { asciiMathToTex, MathFormula } from "../MathFormula";

const formulae = `26 + (this (is) weird) / (45 (why not)) + 3

 46.7 / 65.2 * 6

47 * (this should be)/okok

(1 / 3) /4 * 7 * 8

 4 / (1 / 3) * 10

6 / (sref)

6 / 3

4 / 10 * Sref * A - 20% * (4 / 10 * Sref * A) + 3 * Sref

20% - 30%

 no no / ok ok ok `;

describe("asciiMathToTex", () => {
  it("gives the correct TeX formula provided a correct ascii input", () => {
    expect(asciiMathToTex(formulae))
      .toMatch(`26 + \\frac{(this (is) weird)}{(45 (why not))} + 3

 \\frac{46.7}{65.2} \\times 6

47 \\times \\frac{(this should be)}{okok}

\\frac{(\\frac{1}{3})}{4} \\times 7 \\times 8

 \\frac{4}{(\\frac{1}{3})} \\times 10

\\frac{6}{(sref)}

\\frac{6}{3}

\\frac{4}{10} \\times Sref \\times A - 20\\% \\times (\\frac{4}{10} \\times Sref \\times A) + 3 \\times Sref

20\\% - 30\\%

 no \\frac{no}{ok} ok ok `);
  });
});
describe("<MathFormula />", () => {
  it("renders", () => {
    const { container } = render(
      <MathFormula formula="1 / 4 * Sref * A * 2" />,
    );
    expect(container).toMatchSnapshot();
  });
});
