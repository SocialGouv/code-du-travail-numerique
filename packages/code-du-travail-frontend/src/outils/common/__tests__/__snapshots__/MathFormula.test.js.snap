// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MathFormula /> renders 1`] = `
<div>
  <div
    class="sc-kTCsyW ihNUfD"
  >
    <div
      style="overflow: auto;"
    >
      <span
        class="katex-display"
      >
        <span
          class="katex"
        >
          <span
            class="katex-mathml"
          >
            <math
              display="block"
              xmlns="http://www.w3.org/1998/Math/MathML"
            >
              <semantics>
                <mrow>
                  <mfrac>
                    <mn>
                      1
                    </mn>
                    <mn>
                      4
                    </mn>
                  </mfrac>
                  <mo>
                    ×
                  </mo>
                  <mi>
                    S
                  </mi>
                  <mi>
                    r
                  </mi>
                  <mi>
                    e
                  </mi>
                  <mi>
                    f
                  </mi>
                  <mo>
                    ×
                  </mo>
                  <mi>
                    A
                  </mi>
                  <mo>
                    ×
                  </mo>
                  <mn>
                    2
                  </mn>
                </mrow>
                <annotation
                  encoding="application/x-tex"
                >
                  \\frac{1}{4} \\times Sref \\times A \\times 2
                </annotation>
              </semantics>
            </math>
          </span>
          <span
            aria-hidden="true"
            class="katex-html"
          >
            <span
              class="base"
            >
              <span
                class="strut"
                style="height:2.0074em;vertical-align:-0.686em;"
              />
              <span
                class="mord"
              >
                <span
                  class="mopen nulldelimiter"
                />
                <span
                  class="mfrac"
                >
                  <span
                    class="vlist-t vlist-t2"
                  >
                    <span
                      class="vlist-r"
                    >
                      <span
                        class="vlist"
                        style="height:1.3214em;"
                      >
                        <span
                          style="top:-2.314em;"
                        >
                          <span
                            class="pstrut"
                            style="height:3em;"
                          />
                          <span
                            class="mord"
                          >
                            <span
                              class="mord"
                            >
                              4
                            </span>
                          </span>
                        </span>
                        <span
                          style="top:-3.23em;"
                        >
                          <span
                            class="pstrut"
                            style="height:3em;"
                          />
                          <span
                            class="frac-line"
                            style="border-bottom-width:0.04em;"
                          />
                        </span>
                        <span
                          style="top:-3.677em;"
                        >
                          <span
                            class="pstrut"
                            style="height:3em;"
                          />
                          <span
                            class="mord"
                          >
                            <span
                              class="mord"
                            >
                              1
                            </span>
                          </span>
                        </span>
                      </span>
                      <span
                        class="vlist-s"
                      >
                        ​
                      </span>
                    </span>
                    <span
                      class="vlist-r"
                    >
                      <span
                        class="vlist"
                        style="height:0.686em;"
                      >
                        <span />
                      </span>
                    </span>
                  </span>
                </span>
                <span
                  class="mclose nulldelimiter"
                />
              </span>
              <span
                class="mspace"
                style="margin-right:0.2222em;"
              />
              <span
                class="mbin"
              >
                ×
              </span>
              <span
                class="mspace"
                style="margin-right:0.2222em;"
              />
            </span>
            <span
              class="base"
            >
              <span
                class="strut"
                style="height:0.8889em;vertical-align:-0.1944em;"
              />
              <span
                class="mord mathnormal"
                style="margin-right:0.05764em;"
              >
                S
              </span>
              <span
                class="mord mathnormal"
              >
                re
              </span>
              <span
                class="mord mathnormal"
                style="margin-right:0.10764em;"
              >
                f
              </span>
              <span
                class="mspace"
                style="margin-right:0.2222em;"
              />
              <span
                class="mbin"
              >
                ×
              </span>
              <span
                class="mspace"
                style="margin-right:0.2222em;"
              />
            </span>
            <span
              class="base"
            >
              <span
                class="strut"
                style="height:0.7667em;vertical-align:-0.0833em;"
              />
              <span
                class="mord mathnormal"
              >
                A
              </span>
              <span
                class="mspace"
                style="margin-right:0.2222em;"
              />
              <span
                class="mbin"
              >
                ×
              </span>
              <span
                class="mspace"
                style="margin-right:0.2222em;"
              />
            </span>
            <span
              class="base"
            >
              <span
                class="strut"
                style="height:0.6444em;"
              />
              <span
                class="mord"
              >
                2
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
